### PR TITLE
ci: add SCCACHE_S3_KEY_PREFIX

### LINKED ISSUE #29366: sccache cache wrong binary
#### Problem

https://discord.com/channels/428295358100013066/967028962746327060/1033129370052739112
https://discord.com/channels/428295358100013066/560503042458517505/1055222585106903165

seems it happens when we upgrade our sbf-tools

#### Proposed Solution

find a way to cache them correctly

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

https://discord.com/channels/428295358100013066/560503042458517505/1083170016163274812

#### Summary of Changes

add `SCCACHE_S3_KEY_PREFIX` to share cache in a more stable way. It will create a folder for us. for now I use 
`SCCACHE_S3_KEY_PREFIX="${rust_stable}_${rust_nightly}_${SBF_TOOLS_VERSION}"`

it looks like this on s3
Screenshot 2023-03-09 at 7 01 48 PM

Close #29366

### CODE REVIEW

File: .buildkite/hooks/pre-command [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -e
 
-eval "$(ejson2env .buildkite/env/secrets.ejson)"
```
Comment (yihau) [inline .buildkite/hooks/pre-command:4]: I think we don't use it for a long time. I comment out this line at this PR. I'm going to have another PR to clean this old mechanism

File: .buildkite/hooks/pre-command [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+# eval "$(ejson2env .buildkite/env/secrets.ejson)"
```
Comment (ryoqun) [inline .buildkite/hooks/pre-command:4]: seems unrelated change? (also how about removing the dir as well?)

Comment (ryoqun) [reply .buildkite/hooks/pre-command:4]: ah, I'd prefer to do this as separate pr.

Comment (yihau) [reply .buildkite/hooks/pre-command:4]: I planned to open another PR for it. maybe I make it happen in this PR as well

Comment (yihau) [reply .buildkite/hooks/pre-command:4]: okay! I will open it when this one merge 👻

**Comment by yihau** [comment]
🪖

File: .buildkite/hooks/pre-command (hunk 2 above)
Comment (yihau) [reply .buildkite/hooks/pre-command:4]: the another PR is here https://github.com/solana-labs/solana/pull/30653 🪖
